### DISCUSSION TITLE
ESP32 to MQTT

### DISCUSSION THREAD

**Discussion by techdotrom** [syssi/esphome-jk-bms#154] [Q&A]
Hi,
Would anyone be willing to do a full tutorial on how to get this working on Home Assistant or MQTT?

I've spent an unreasonably long time trying to get this working on an ESP32 with Home Assistant and without Home Assistant but trying to use my MQTT broker instead. 

I'm very familiar with the Arduino IDE, grafana, influx etc... Home Assistant and ESP home is blowing my mind, I just cant seem to get my head around it.

I got to the stage where I can hear the BMS beep when the ESP is next to it (esp32 ble) but then nothing? nothing shows on MQTT explorer and nothing shows on Home Assistant. 

I currently run home assistant on a VM but I will be installing it on a dedicated computer at some point - it looks like it can do everything I do now across 3 computers all on one.  

I've got to the stage where I know I'm missing something in the process but I don't know what. 

Any help or pointers would be hugely appreciated!

Thanks.

**Comment by syssi** [comment]
If you provide the log of your ESPHome node I could try to provide some support / feedback. The BMS beeps on every received command. Beeping is a good sign! The ESP is able to talk to the BMS and the log should show some traffic already.

If the status LED of the BMS blinks: No BLE connection established.
If the status LED blinks: BLE connection active!

The BMS supports just one BLE connection at a time.

**Reply by techdotrom** [reply]
I've just managed to get it onto my mqtt (I was adding the mqtt port after the IP which you don't need to do). - Can't believe that's what it was. 2 days I've been messing about with this!!!

One other thing I would like to ask which I don't understand - If I want to add the ESP to ESPHome on Home Assistant, how do I get the "api" from Home Assistant to put in the YAML code on the ESP?

Thanks very much.

**Reply by syssi** [reply]
If you add the line `api:` to your configuration yaml and flash the node again (`esphome run config.yaml`) Home Assistant should be able to discover the node automatically (as discovery notification on the left). If HA doesn't show this notification you can go to the `ESPHome integration` (Settings -> Devices -> Add integration -> ESPHome -> "hostname.local") to add the node using the local mDNS name manually. You can remove the `mqtt` section if you don't need MQTT for other applications afterwards.
